### PR TITLE
Example markup test: DO NOT MERGE

### PR DESCRIPTION
**Pull request by kwalrath**
This is a first attempt at marking up some example code in the format proposed by @petebacondarwin.

### CODE REVIEW

File: modules/angular2/src/core/annotations/annotations.js [hunk 1]
```diff
@@ -59,56 +59,90 @@ import {Injectable} from 'angular2/di';
...
- * ```
- *
- * Let's step through the different ways in which `MyDirective` could be declared...
+ * The following example demonstrates how dependency injection resolves
+ * constructor arguments in practice.
+ * The [ app.ts | app.js | main.dart ] file contains
+ * a `dependency` decorator and `SomeService` injectable class.
+ * `index.html` contains the HTML template for the `dependency` decorator.
+ *
+ * [PENDING: Does order matter when including the files? Should it be consistent?
```
Comment (petebacondarwin) [inline modules/angular2/src/core/annotations/annotations.js]: In Angular1 the order matters as that this order that the tabs appear in the doc.

File: modules/angular2/src/core/annotations/annotations.js [hunk 2]
```diff
@@ -59,56 +59,90 @@ import {Injectable} from 'angular2/di';
...
+ * Can it or the selected tab vary so that the most important file is visible first?]
+ * [NOTE: It's hard to visually distinguish HTML code from its container.]
+ * [PENDING: how to refer to files when the names differ in TS/ES/Dart
+ * (e.g., when the file suffix is the only different, or when the actual name
+ * is different?]
```
Comment (petebacondarwin) [inline modules/angular2/src/core/annotations/annotations.js]: How about we have three attributes on a file:

* name - this is a unique identifier for this file
* type - this is an optional attribute that specifies the type of file (i.e. html, dart, js, ts, etc.) If not given then we try to guess it from the extension of the `name` attribute.
* filename - this is an optional attribute that specifies the name of this file, if not given then we try to guess it from the `name` and `type` attributes.

File: modules/angular2/src/core/annotations/annotations.js [hunk 3]
```diff
@@ -59,56 +59,90 @@ import {Injectable} from 'angular2/di';
...
+ * [PENDING: it'd be great to transpile to Dart, as a rule, but how should I
+ * indicate that something should NOT be transpiled because I have a more
+ * idiomatic way of doing it?]
```
Comment (petebacondarwin) [inline modules/angular2/src/core/annotations/annotations.js]: How about we have a fourth attribute `transpile`, which we apply to files that should be transpiled from another file and contains the name of the file to transpile. So if we have a `sample.ts` file and we want a `sample.dart` file, we could provide:

```html
<file name="sample" type="ts">Some generic TS code</file>
<file name="sample" type="dart" transpile="sample.ts"></file>
```

Note that the second file element would be empty as we would get contents by transpiling the TypeScript file. We may need to provide additional attributes for how the transpilation should happen.

File: modules/angular2/src/core/annotations/annotations.js (hunk 1 above)
Comment (kwalrath) [inline modules/angular2/src/core/annotations/annotations.js]: That makes sense. In the next version I'll add some questions about how to deal with tab order vs. readability.

File: modules/angular2/src/core/annotations/annotations.js (hunk 2 above)
Comment (kwalrath) [inline modules/angular2/src/core/annotations/annotations.js]: OK, sounds good.

File: modules/angular2/src/core/annotations/annotations.js (hunk 3 above)
Comment (kwalrath) [inline modules/angular2/src/core/annotations/annotations.js]: I'm a little worried about what will happen if there's no type="dart" file. Would there be no example for Dart users, or a TS example for Dart users? I think I'd prefer the latter, but I wonder if automatically transpiling the code (unless a "dart" version is specified) might be better.

**Comment by kwalrath** [comment]
OK, @petebacondarwin I've uploaded a version that does what I think you proposed.

**Comment by mhevery** [comment]
Can we close this PR? If you want a place to discuss changes, may I suggest creating a PR against your own repo? (if you need help on how to do that, happy to help.)

**Comment by kwalrath** [comment]
@mhevery sure, I can do that.

**Comment by kwalrath** [comment]
Moved it to https://github.com/kwalrath/angular/pull/1.

**Pull request closed without merging**